src/Run/CommandLine.cs

```diff
@@ -0,0 +1,1876 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Execute;
+
+using System.Reflection;
+
+namespace Microsoft.Fx.CommandLine
```

Comment:
I'm using a great part of it. The methods I'm not using is because I don't have positional parameters or I'm not working with enums.
I also though about making the changes in the core version but there are changes that are too specific for this context. 
We could prune it to make it cleaner